scripts/uploadToS3.js

```diff
@@ -32,11 +32,22 @@ const files = glob.sync(options.files, {
 })
 
 const generateHeaders = file => {
-  const contentType = mime.getType(path.extname(file))
+  const extension = path.extname(file)
+  const contentType = mime.getType(extension)
+  const shouldGzip = [".js", ".css", ".gz"].some(ext => ext === extension)
+
+  let optionalHeaders = {}
+  if (shouldGzip) {
+    optionalHeaders = {
+      ContentEncoding: "gzip",
+    }
+  }
+
   return {
+    ACL: "public-read",
     CacheControl: "max-age=315360000, public",
     ContentType: contentType,
-    ACL: "public-read",
+    ...optionalHeaders,
```

Comment:
Okay, so walk me through this... So we're uploading the gzipped files (only?) to S3, and then S3 (or cloudfront) handles pulling the right file with the right header when the related js files are requested?

Reply:
We're uploading all assets from `public` folders to S3. If the file matches a certain extension (js, css, or gz), add the `ContentEncoding: gzip` header. Technically CloudFront should automatically do this if it detects a certain type of file, but it seems inconsistent. I'm forcing it here to see if this might stabilize things a bit. (Checking in my test bucket ContentEncoding seems properly applied when inspecting Metadata.)